compiler/cpp/src/generate/t_cpp_generator.cc

```diff
@@ -1626,6 +1640,16 @@ void t_cpp_generator::generate_struct_ostream_operator_decl(std::ofstream& out,
   out << "std::ostream& operator<<(std::ostream& out, const " << tstruct->get_name() << "& obj)";
 }
 
+void t_cpp_generator::generate_exception_what_method_decl(std::ofstream& out,
+                                                          t_struct* tstruct,
+                                                          bool external) {
+  std::string prefix = "";
+  if (external) {
+    prefix = string(tstruct->get_name() + "::");
+  }
+  out << "const char* " << prefix << "what() const throw()";
```

Comment:
throw() is deprecated (especially that you concatenate strings there and you cannot guarantee "no throw" behavior)